core/src/main/java/org/elasticsearch/action/fieldstats/FieldStats.java

```diff
@@ -550,27 +625,50 @@ public static FieldStats readFrom(StreamInput in) throws IOException {
         long sumTotalTermFreq = in.readLong();
         boolean isSearchable = in.readBoolean();
         boolean isAggregatable = in.readBoolean();
-
+        boolean hasMinMax = true;
```

Comment:
I am not sure of this, NamedWriteable is ok when you want to serialize objects that have nothing in common but that's not the case here. We just want to serialize the min/max value differently for each specialization of the class. Rewriting this as a NamedWriteable would be completely different. We would need to create a class to encapsulate the min/max info and then use NamedWriteable to serialize these classes. I can do that but I prefer to do it in another PR since it's not related to the change that we're trying to make here ?